lib/srv/alpnproxy/local_proxy_test.go

```diff
@@ -533,12 +533,18 @@ func TestKubeMiddleware(t *testing.T) {
 		err := km.CheckAndSetDefaults()
 		require.NoError(t, err)
 
-		rw := responsewriters.NewMemoryResponseWriter()
-		// HandleRequest will reissue certificate if needed.
-		km.HandleRequest(rw, req)
-
-		// request timed out.
-		require.Equal(t, http.StatusInternalServerError, rw.Status())
+		var rw *responsewriters.MemoryResponseWriter
+		// We use `require.Eventually` to avoid a very rare test flakiness case when reissue goroutine manages to
+		// successfully finish before the parent goroutine has a chance to check the context (and see that it's expired).
+		require.Eventually(t, func() bool {
+			rw = responsewriters.NewMemoryResponseWriter()
+			// HandleRequest will reissue certificate if needed.
+			km.HandleRequest(rw, req)
+
+			// request timed out.
+			return rw.Status() == http.StatusInternalServerError
+
+		}, time.Second, 100*time.Millisecond)
```

Comment:
I would wait for 3-5 seconds. 1 second rarely seems long enough for us.

Reply:
This happens so rarely I think 10 attempts would be enough, but no harm in increasing the timeout to 5 seconds anyway - done.